src/Microsoft.Identity.Web/MigrationAid/Obsolete-WebApiAuthenticationBuilderExtensions.cs

```diff
@@ -36,11 +36,14 @@ public static partial class WebApiAuthenticationBuilderExtensions
             X509Certificate2 tokenDecryptionCertificate = null,
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
         {
-            return builder.AddMicrosoftWebApi(
-                configuration,
-                configSectionName,
-                jwtBearerScheme,
+            // Just call the obsolete method below (which takes delegates).
+            // This method will do the work of taking into account the legacy
+            // parameter for the token decyrption certificate
+            return builder.AddProtectedWebApi(
```

Comment:
The principle is to not call immediately the AddMicrosoftWebApi method but delegate to an obsolete method which will handle the legacy certificate.